src/Modules/LearningTasks/Tutor.LearningTasks.Core/UseCases/Authoring/ActivityService.cs

```diff
@@ -0,0 +1,64 @@
+﻿using AutoMapper;
+using FluentResults;
+using Tutor.BuildingBlocks.Core.UseCases;
+using Tutor.LearningTasks.API.Dtos.Activities;
+using Tutor.LearningTasks.API.Public;
+using Tutor.LearningTasks.API.Public.Authoring;
+using Tutor.LearningTasks.Core.Domain.Activities;
+using Tutor.LearningTasks.Core.Domain.RepositoryInterfaces;
+
+namespace Tutor.LearningTasks.Core.UseCases.Authoring;
+
+public class ActivityService : CrudService<ActivityDto, Activity>, IActivityService
+{
+    private readonly IActivityRepository _activityRepository;
+    private readonly IAccessServices _accessServices;
+
+    public ActivityService(IActivityRepository activityRepository, IAccessServices accessServices,
+        ILearningTasksUnitOfWork unitOfWork, IMapper mapper) : base(activityRepository, unitOfWork, mapper)
+    {
+        _activityRepository = activityRepository;
+        _accessServices = accessServices;
+    }
+
+    public Result<List<ActivityDto>> GetByCourse(int courseId)
+    {
+        List<Activity> activities = _activityRepository.GetCourseActivities(courseId);
+        return MapToDto(activities);
+    }
+
+    public Result<ActivityDto> Create(ActivityDto activity, int instructorId)
+    {
+        if (!_accessServices.IsCourseOwner(activity.CourseId, instructorId))
+            return Result.Fail(FailureCode.Forbidden);
+
+        if (activity.Subactivities?.Exists(subactivity => Get(subactivity.ChildId).IsFailed) == true)
```

Comment:
I try to check if Activity with id exists in database. Maybe in subactivities is id of Activity that does not exist.

Reply:
You are making the assumption here that the supplied ID might be invalid. However, the only check that you are actually making is if the ID exists, not if it is valid.

To illustrate, say that our intent is to create a subactivity for Activity with the ID 5. Your code does not validate if something other than 5 is supplied, only that the ID exists in the DB.

I would argue that we need to handle this case when the activity does not exist in the READ operations, since there is a greater chance that an activity can be deleted after the LT and subactivities are created (either through a use case or by mistake). The front-end should display "Data aktivnost ne postoji" or something along those lines for Steps and Subactivities that cannot be retrieved from the DB.